apps/safe-claiming-app/src/widgets/ClaimingWidget.tsx

```diff
@@ -0,0 +1,213 @@
+        sameAddress(entry.address, delegateAddressFromContract)
+      )
+      return (
+        registeredDelegateFromData || { address: delegateAddressFromContract }
+      )
+    }
+  }, [delegateAddressFromContract, delegates])
+
+  const [userVestingStatus, , userVestingLoading] = useFetchVestingStatus(
+    userVesting?.vestingId,
+    userVesting?.contract
+  )
+
+  const [ecosystemVestingStatus, , ecosystemVestingLoading] =
+    useFetchVestingStatus(
+      ecosystemVesting?.vestingId,
+      ecosystemVesting?.contract
+    )
+
+  const [investorVestingStatus, , investorVestingLoading] =
+    useFetchVestingStatus(investorVesting?.vestingId, investorVesting?.contract)
+
+  const totalClaimed = BigNumber.from(userVestingStatus?.amountClaimed || 0)
+    .add(ecosystemVestingStatus?.amountClaimed || 0)
+    .add(investorVestingStatus?.amountClaimed || 0)
+
+  const votingPower = totalAllocation.add(balance || 0).sub(totalClaimed)
+
+  const unredeemedAllocations =
+    (!userVestingStatus?.isRedeemed &&
+      BigNumber.from(userVesting?.amount || 0).gt(0)) ||
+    (!ecosystemVestingStatus?.isRedeemed &&
+      BigNumber.from(ecosystemVesting?.amount || 0).gt(0))
+
+  const ctaWidget = (
+    <>
+      <div>
+        <Title>Become the part of Safe future!</Title>
+        <br />
+        <Subtitle>
+          You will be able to buy $SAFE once the token transferability is
+          enabled.
+        </Subtitle>
+      </div>
+      <StyledButton disabled>Buy Tokens</StyledButton>
+    </>
+  )
+
+  const votingPowerWidget = (
+    <>
+      <div>
+        <Title>Your voting power</Title>
+        <Box
+          display="flex"
+          alignItems="center"
+          justifyContent="center"
+          gap={1}
+          mb={1}
+        >
+          <SafeIcon />
+          <Typography variant="h5">
+            {formatAmount(Number(ethers.utils.formatEther(votingPower)), 2)}{" "}
+          </Typography>
+        </Box>
+      </div>
+      {totalClaimed.gt(0) ? (
+        <>
+          <Subtitle>
+            You've already claimed{" "}
+            {formatAmount(Number(ethers.utils.formatEther(totalClaimed)), 2)}{" "}
+            SAFE
+          </Subtitle>
+          {currentDelegate && (
+            <Box width={1}>
+              <Typography variant="caption" marginBottom={1}>
+                Delegated to
+              </Typography>
+              <SelectedDelegate delegate={currentDelegate} onClick={() => {}} />
+            </Box>
+          )}
+        </>
+      ) : (
+        <>
+          {unredeemedAllocations && (
+            <Subtitle>
+              You have unredeemed tokens. Redeem them by 27th Dec or they will
+              be transfered back into the treasury.
+            </Subtitle>
+          )}
+          <Link
+            href={`${window.location.ancestorOrigins[0]}/apps?safe=${
+              safe.chainId === 1 ? "eth" : "gor"
+            }:${safe.safeAddress}&appUrl=${CLAIMING_APP_URL}`}
```

Comment:
I would suggest we move this out of render, and the `chainId` ternary to a constant.